src/Essential.Diagnostics.RollingFileTraceListener/Diagnostics/RollingFileTraceListener.cs

```diff
@@ -111,6 +112,36 @@ public bool ConvertWriteToEvent
             }
         }
 
+        private bool? _newStreamOnError = null;
+        /// <summary>
+        /// Gets or sets whether errors writing to the file should cause a new file stream to be instantiated.
+        /// Useful when the drive containing the file has a transient fault (usb stick removed and reinserted, network outage of mapped drive, etc.)
+        /// </summary>
+        [SuppressMessage("Microsoft.Usage", "CA1806:DoNotIgnoreMethodResults", MessageId = "System.Boolean.TryParse(System.String,System.Boolean@)", Justification = "Default value is acceptable if conversion fails.")]
+        public bool NewStreamOnError
+        {
+            get
+            {
+                if (_newStreamOnError.HasValue)
+                {
+                    return _newStreamOnError.Value;
+                }
+
+                // Default behaviour is to convert Write to event.
+                var newStreamOnError = false;
+                if (Attributes.ContainsKey("newStreamOnError"))
+                {
+                    bool.TryParse(Attributes["newStreamOnError"], out newStreamOnError);
+                }
+                _newStreamOnError = newStreamOnError;
+                return newStreamOnError;
+            }
+            set
+            {
+                Attributes["newStreamOnError"] = value.ToString(CultureInfo.InvariantCulture);
```

Comment:
You store it in a backing field, but don't update it on set. Don't need the backing field.

Reply:
Done, I made the setter set the backing field.